### DISCUSSION TITLE
Comparing HSGQ and ODI xPON Stick

### DISCUSSION THREAD

**Discussion by iodize6399** [Anime4000/RTL960x#148] [Q&A] [answered]
I recently placed an order on [Alibaba](https://hsgq.en.alibaba.com/) for an HSGQ xPON Stick from Shenzhen HS Fiber Communication Equipment Co., Ltd, but I am still awaiting its delivery. According to the sales representative, this xPON device is identical to the one produced by ODI. The rep also claimed it's the RTL9601D variant.

The sales rep also informed me that ODI is a sister company of HSGQ. I corroborated this claim by comparing the MAC OUI of both devices and found them to be the same. 

Now, I am curious to know if the firmware available on this repository for ODI's xPON can also be used with the HSGQ variant. Does anyone else have any experience with the HSGQ xPON Stick?

![hsgq-xpon](https://user-images.githubusercontent.com/4197707/231159152-8b7b0487-30d2-4bff-b150-cec78bc927c3.PNG)

**Comment by iodize6399** [comment, marked as answer]
Just received this device and the firmware is interestingly new than available on the repo. Other than that and webux, everything remains the same.
![image](https://user-images.githubusercontent.com/4197707/235885728-a6d443eb-f4ea-4ed9-89a8-1bf92584a2dd.png)

**Reply by Anime4000** [reply]
Nice, it look same as the interface
If you can extract V1.2.2-221209 firmware, would be nice to keep

**Reply by iodize6399** [reply]
Few observations regarding my experience dealing with HSGQ:

- Changing `OMCI_SW_VER1` and `OMCI_SW_VER2` will not stick on firmware version `v221209`, despite it being newer (#30).
- The HSGQ support representative has therefore advised downgrading to `M110_sfp_ODI_220923.tar`, which is also available on the repository. In fact, they uploaded an ODI firmware for me with the same name, therefore I'm now 100% confident in assuming they are the same company that shares the exact same hardware.
- Personally, I purchased the `HSGQ DFP-34X-2C2` directly from their Alibaba store for approximately $32, and I found their support staff, particularly Emily Yuen, to be quite helpful and responsive. In contrast, based on some comments I've seen here, dealing with Fibershow can be quite painful. I believe Firbershaw is a reseller.
- When asked about the .tar for `v221209` she said, "Later will add it in next firmware", it's not entirely clear what that means. I'll try reminding her about this after a week again.
- She also confirmed that `v221209` is **HGU** mode.
- Unfortunately, attempting to extract the firmware files using `scp admin@192.168.1.1:/mtd13 "D:\xponbackup\mtd13"` on a Windows machine did not work, as a timeout error message was received `([timeout](ssh: connect to host 192.168.1.1 port 22: Connection refused))`. If anyone can assist with pulling the firmware using scp from a Windows machine to extract v221209, I would happily wait before downgrading to `v220923`.

**Reply by iodize6399** [reply]
Well, they provided me with .tar for the newer firmware `v221209` after a while.

SHA 256 `94348ff82019562ab0d10814d307f5594bc8c4108676e196aa34fe1c413a3c15`
Mode `IGD/HGU`
Filename `M114_sfp_ODI_hybrid_221209.tar`

[M114_sfp_ODI_hybrid_221209.zip](https://github.com/Anime4000/RTL960x/files/11396084/M114_sfp_ODI_hybrid_221209.zip)

**Reply by Anime4000** [reply]
great, I add into repo

**Reply by iodize6399** [reply]
They just sent me an official DFP-34X-2C2 user manual albeit in Chinese.

SHA 256 `aa3321be3abd70ed0abf727a1b30f768d0cfb7e2d3aa90078738d0c1234ead5d`

[XPON Stick 替换运营商光猫解决方案_V2 0.pdf](https://github.com/Anime4000/RTL960x/files/11404592/XPON.Stick._V2.0.pdf)

**Reply by Anime4000** [reply]
these look like Guide how to clone info

**Reply by Blacktimess** [reply]
Worthwhile going from 220923 to 221209 for ODI? Going from SFU to HGU or loosing port emu is not a concern.

**Reply by Blacktimess** [reply]
Will answer myself, have dual boot with 220923 and 221209, both working perfectly.

**Reply by kevindd992002** [reply]
How do you dual boot with the DFP-34X-2C2?

**Reply by rndm2** [reply]
It looks like 221209 works better: consume less cpu and mem. VLANs at the same time works fine too.

**Reply by jvcell** [reply]
Depois que instalei a versão 221209, não consigo mais acessar pelo putty via ssh, so consigo se fizer o downgrade para a verão 220923, porem nessa verão a VLAN para mim não funciona direitinho, alguém poderia me ajudar? alguém tem o mesmo problema e conseguiu resolver? nas configurações todas opções ftp e demais estão ativas.

**Comment by iodize6399** [comment]
I found HSGQ's official Taobao store: https://shop143937166.taobao.com/shop/view_shop.htm

They are selling both [ODI](http://a.m.taobao.com/i673661619814.htm) (at a high price) and [HSGQ](http://a.m.taobao.com/i675179133761.htm), when in fact both have the same underlying hardware.

**Comment by Blacktimess** [comment]
Hi Kevin,

To check active image: nv getenv sw_active

To change to image 0: nv setenv sw_commit 0
Reboot

To change to image 1: nv setenv sw_commit 1
Reboot

See: ODI Realtek DFP-34X-2C2 | Hack GPON (hack-gpon.org)
<https://hack-gpon.org/ont-odi-realtek-dfp-34x-2c2/>

**Comment by ZCAT7788** [comment]
似乎PPPOE拨号下端口转发有问题,无论怎么设置都无法起作用~

**Reply by Anime4000** [reply]
在 GPON Stick 内使用 PPPoE？

**Reply by ZCAT7788** [reply]
是的,插在交换机上当路由用了....哈哈
我用尽了各种办法,都没让端口转发正常运行..

**Reply by ZCAT7788** [reply]
现在PPPOE以及其他功能都正常了,现在就是端口转发怎么设置都没法起作用,就是打不开我自己的NAS域名,之前用光猫和单独使用路由设置端口转发后都正常用打开NAS域名~

**Reply by ZCAT7788** [reply]
我试过后面两版固件都不行..

**Reply by NekokeCore** [reply]
是不是没有本地环回？用流量试试，顺带问一下，HGU模式下IPV6正常吗，IGMP代理可以设置组播VLAN吗

**Comment by Strykar** [comment]
I just bought two of these, arrived today!
![image](https://github.com/Anime4000/RTL960x/assets/2946372/7c6f3636-df9f-4abc-92e8-a328c1c26f7e)

Here's the english user manual: https://github.com/Strykar/GPON/blob/main/XPON%20Sticker%20User%20Manual_V2.0.pdf
And a spec sheet with the pinouts:  https://github.com/Strykar/GPON/blob/main/PON%20stick%20Spec.pdf
